components/script/body.rs

```diff
@@ -891,22 +896,25 @@ fn run_form_data_algorithm(
     Err(Error::Type("Inappropriate MIME-type for Body".to_string()))
 }
 
-#[allow(unsafe_code)]
+fn run_bytes_data_algorithm(cx: JSContext, bytes: Vec<u8>) -> Fallible<FetchedData> {
```

Comment:
Note: I think this check rather belongs in `get_read_promise_bytes`, and we should restructure the algorithm so that it isn't tied to fetch only, but rather calls into an implementation of [`read all bytes `](https://streams.spec.whatwg.org/#readablestreamdefaultreader-read-all-bytes) that will be provided with https://github.com/servo/servo/pull/35151

Good for a follow-up depending on that PR. 

 cc @yoseio